Code/Tools/AssetProcessor/native/AssetDatabase/AssetDatabase.cpp

```diff
@@ -1117,10 +1120,19 @@ namespace AssetProcessor
             if (m_databaseConnection->ExecuteOneOffStatement(CREATE_STATS_TABLE))
             {
                 foundVersion = DatabaseVersion::AddedStatsTable;
-                AZ_TracePrintf(AssetProcessor::ConsoleChannel, "Upgraded Asset Database to version %i (AddedStatsTable)\n", foundVersion)
+                AZ_TracePrintf(AssetProcessor::ConsoleChannel, "Upgraded Asset Database to version %i (AddedStatsTable)\n", foundVersion);
             }
         }
 
+        if(foundVersion == DatabaseVersion::AddedStatsTable)
+        {
+            // Version update - change SourceDependency Source to SourceGuid column
+            // Do nothing so the whole database is dropped.
+            // Unfortunately we have to reprocess all assets because of the way the fingerprinting algorithm works,
+            // changing from storing the path to the UUID changes the fingerprint, resulting in all assets reprocessing anyway
+            AZ_TracePrintf(AssetProcessor::ConsoleChannel, "Asset database version updated to ChangedSourceDependencySourceColumn, database will be cleared as migration is not possible for this update\n", foundVersion);
```

Comment:
Unclear what database will be cleared means, what are we trying to tell the user?

Reply:
This is probably geared more toward technical users since this isn't going to show up in the GUI, its just saying the database will be erased, which implies all the assets need to be rebuilt.  I guess I can expand the message, but its really just a note for potential debugging.